projects/test_project/rules.mk

```diff
@@ -1,8 +1,6 @@
-# Specify the device family to build for
-DEVICE_FAMILY := stm32f0xx
-
 # Specify the main file (relative to this file if not in this folder)
 MAIN := main.c
 
 # Specify the device library you want to include
-DEVICE_LIBRARY := libstm32f0.a
+DEPS := libraries/stm32f0xx/stm32f0 \
```

Comment:
Sure, I'll give you a couple: let's say I create a wrapper for something we have in ms-lib from the last car an example of this was the pwm current calculation wrapper for the ring ring_buffer from the last car. I can package that wrapped module as a library and include it in the plutus build. The pwm pins timers and interrupts were different for the two boards I tested it on so by creating a modified implementation and including it for both cases and just modifying the config would have meant I didn't need to constantly change the code back and forth. They share the core components and just have a modified config. Meanwhile ms-lib remains a general library whereas plutus has its own version of the library for that functionality that can support different revisions or testing setups.

Similarly if we wanted a special implementation of the FSM for one board it can be created as a separate library without having to put that library in ms-lib. Another benefit would be to create a shared library for the lights boards, if we do have two implementations they can share a library of basic functionality that is specific to lights but the configurations of the two boards can be different. No-one else cares about this but the light boards both do. 

This is also nice because you can unit test a library as a separate module and then integrate it. It encourages more modular/organized code and unit testing a library means it is nicely reusable even if the core code gets refactored.